exporters/otlp/CMakeLists.txt

```diff
@@ -113,8 +113,9 @@ if(WITH_OTLP_HTTP)
 
   target_link_libraries(
     opentelemetry_exporter_otlp_http_client
-    PUBLIC opentelemetry_sdk opentelemetry_proto opentelemetry_http_client_curl
-           nlohmann_json::nlohmann_json)
+    PUBLIC opentelemetry_sdk
```

Comment:
I think it would be better to public link to `opentelemetry_ext`, because `opentelemetry_exporter_otlp_http_client` use the headers of http client.

Reply:
Yes, opentelemetry_ext is a interface target.

Reply:
So you'd like me to add it as a dependency even though it wasn't there before I made my change - I can do that, but it seems a bit unfair to hold up my PR because of a previously missing dependency.

Reply:
Sorry for that, but we did not previously miss this dependency. The reason for this is that `opentelemetry_ext` is publicly linked to `opentelemetry_http_client_curl` in https://github.com/open-telemetry/opentelemetry-cpp/blob/main/ext/src/http/client/curl/CMakeLists.txt. Additionally, `opentelemetry_ext` will be implicitly publicly linked by `target_link_libraries(opentelemetry_exporter_otlp_http_client PUBLIC opentelemetry_http_client_curl)`. However, after we privately link `opentelemetry_http_client_curl` to `opentelemetry_exporter_otlp_http_client`, the dependency will become invisible to other targets that use `target_link_libraries(<TARGET_NAME> PUBLIC|PRIVATE opentelemetry_exporter_otlp_http_client )`. 
This change may break certain use cases.
*Currently, we lack unit tests for this type of scenario. I think we can add these tests in the future and in other PRs and issues.*

Reply:
Ok - that's fair enough.
Although I'm a little curious about the fine grained internal parts of OT because doing it this way implies that you could have only some of them instlled on a system, but the extensive internal dependencies seems to me to mean that the options for this are actually much more limited and OT actually in practice tends to be more of a monolith. Albeit with some exceptions around the exporters.
I do understand that the intent is that the API itself is entirely header file and standalone, but in practice you always need an implementation and some exporters anyway, and these seem much more monolithic.
Sorry if this is a bit ranty but I'm just finding that working with the OT internal dependencies to be very unintuitive and difficult seeing that it is a large project with lots of internal dependencies that are hiding external ones that are hard to deal with.

Reply:
Yes, we use cmake CONFIG and modern cmake targets to resolve dependencies.Many components are optional but some of them depend each other. We hope users just concerns the components they use and the depended component can be added automatically.We use `target_link_libraries(A PUBLIC B)` in cmake and `deps` setting in bazel.They are similar.